### PR TITLE
[#445] Add gofmt validation to CI

### PR DESCRIPTION
**Pull request by pedroMMM**
Add gofmt validation to CI requested on #445 

##### Checklist

- [x] `make test-all` (UNIX) passes. CI will also test this.

### Description of change

Add a `make fmt` that checks if the code formating is valid. Ensure it's part of the main `make`  invoked by the CI and fail the build if the formating invalid.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -24,6 +24,14 @@ lint:
 	#go tool vet .
 	golint $(pkgs) | grep -v 'unexported' || true
 
+fmt:
+	$(info INFO: Starting build $@)
+	{ \
+set -e ;\
+fmt=$$(gofmt -l .) ;\
+[ -z "$$fmt" ] && echo "valid gofmt" || (echo "invalid gofmt"; exit 1)\
```
Comment (aelsabbahy) [inline Makefile]:
```suggestion
[ -z "$$fmt" ] && echo "valid gofmt" || (echo -e "invalid gofmt\n$$fmt"; exit 1)\
```

When I ran it it just said: "invalid gofmt" without any other details, this adds the actual errors.

Comment (pedroMMM) [reply Makefile]: Great catch!

**Comment by pedroMMM** [comment]
I added both of your suggestions.

**Comment by aelsabbahy** [comment]
`GO_FILES` ignores `*_test.go` files. Is this intentional?

**Comment by pedroMMM** [comment]
> `GO_FILES` ignores `*_test.go` files. Is this intentional?

Yes, analysis tools usually will ignore test files. But with that said, I don't feel strong enough about it so I am open to change it if you want.

**Comment by aelsabbahy** [comment]
Ok, cool. Just wanted to make sure it was intentional. Thanks for this (and all the other work so far)!

**Pull request merged**